src/status_im/ethereum/stateofus.cljs

```diff
@@ -24,12 +25,26 @@
   (when (and name (string/ends-with? name domain))
     (first (string/split name "."))))
 
-(def registrars
+(def old-registrars
   {:mainnet "0xDB5ac1a559b02E12F29fC0eC0e37Be8E046DEF49"
    ;;NOTE: can be enabled for testing builds
    ;;:testnet "0x11d9F481effd20D76cEE832559bd9Aca25405841"
    })
 
+(def registrars-cache (atom {}))
+
+(defn get-registrar [chain callback]
+  (if-let [contract (get @registrars-cache chain)]
+    (callback contract)
+    (let [registry (get ens/ens-registries chain)]
+      (ens/get-addr
```

Comment:
```suggestion
      (ens/get-owner
```

Due the ENS bug, and it's migration to new address, currently the get-addr of mainnet cannot be changed. Although, the owner can achieve the same functionality. 

The use of "get-addr" instead of "get-owner" was chosen as gas optimal for future versions of UsernameRegistrar with https://github.com/status-im/ens-usernames/issues/108 in mind, however it won't block this planned feature or be absurdly more expansive to implement.